### ISSUE TITLE
List available locales we can use with DataFaker

### ISSUE THREAD

**Issue by jonathanlermitage** [datafaker-net/datafaker#566]
Is it possible to list all available locales we can use with DataFaker?  
I would like to pick random locales when I generate data. Also, I developed an [IntelliJ plugin](https://github.com/jonathanlermitage/ij-ilove-devtoys) based on your amazing DataFaker lib, and I'm trying to find a way to show all available locales...  
Thx!

**Comment by bodiam** [comment]
I believe that's "faker.locality.allSupportedLocales()"

**Comment by jonathanlermitage** [comment]
Thx @bodiam!  
It sounds good. It doesn't work for now, but I think this is a bug: code lists locale files with 
```
String[] resourceFiles = new File(./src/main/resources).list();
```
which cannot work outside unit tests. It should use `class.getResource` methods, something like this. Actually, I don't know if we can list files in classpath resources... 🤔

**Comment by bodiam** [comment]
Thanks for reporting this! That's of course a silly bug, sorry about that! I'll see what I can do to fix that.

**Comment by alx-mag** [comment]
In my project I use a Gradle task that writes all available locales into the `resources/dataFakerLocales.txt` file. This file then parsed at runtime. You can use my workaround: https://github.com/alx-mag/lorem/blob/c5c8a5dabf70d0dc5c6a079b14783b855b1391dd/build.gradle.kts#L137.

**Comment by snuyanzin** [comment]
There is an issue with this solution: it does not support locales added with custom providers.

Here is a fix for generic case at #589 please have a look

// cc @jonathanlermitage  as a reporter

**Comment by snuyanzin** [comment]
finally it's merged
`faker.locality().allSupportedLocales());`
now should return the actual list.
Please reopen if it does not work
